### PR TITLE
refactor: ignore handlers if events for an handler isnt provided

### PR DESCRIPTION
**Pull request by AwesomeStickz**

### CODE REVIEW

File: packages/bot/src/handlers/channels/CHANNEL_CREATE.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import type { Bot, DiscordChannel, DiscordGatewayPayload } from '../../index.js'
 
 export async function handleChannelCreate(bot: Bot, payload: DiscordGatewayPayload, shardId: number): Promise<void> {
+  if (bot.events.channelCreate === undefined) return
```
Comment (Skillz4Killz) [inline packages/bot/src/handlers/channels/CHANNEL_CREATE.ts:4]: eww

File: packages/bot/src/handlers/channels/CHANNEL_DELETE.ts [hunk 2]
```diff
@@ -2,10 +2,12 @@ import type { DiscordChannel, DiscordGatewayPayload } from '@discordeno/types'
 import type { Bot } from '../../bot.js'
 
 export async function handleChannelDelete(bot: Bot, data: DiscordGatewayPayload): Promise<void> {
+  if (bot.events.channelDelete === undefined) return
```
Comment (Skillz4Killz) [inline packages/bot/src/handlers/channels/CHANNEL_DELETE.ts:5]: eww

File: packages/bot/src/handlers/channels/CHANNEL_PINS_UPDATE.ts [hunk 3]
```diff
@@ -2,9 +2,11 @@ import type { DiscordChannelPinsUpdate, DiscordGatewayPayload } from '@discorden
 import type { Bot } from '../../bot.js'
 
 export async function handleChannelPinsUpdate(bot: Bot, data: DiscordGatewayPayload): Promise<void> {
+  if (bot.events.channelPinsUpdate === undefined) return
```
Comment (Skillz4Killz) [inline packages/bot/src/handlers/channels/CHANNEL_PINS_UPDATE.ts:5]: eww

**Review by Skillz4Killz — CHANGES_REQUESTED** [review]

**Comment by AwesomeStickz** [comment]
Changed the if conditions, you can now review the PR again @Skillz4Killz

**Review by Skillz4Killz — APPROVED** [review]

**Pull request merged**